astropy/coordinates/representation/base.py

```diff
@@ -505,10 +499,8 @@ def __repr__(self):
         arrstr = np.array2string(self._values, prefix=prefixstr, separator=", ")
 
         diffstr = ""
-        if getattr(self, "differentials", None):
-            diffstr = "\n (has differentials w.r.t.: {})".format(
-                ", ".join([repr(key) for key in self.differentials.keys()])
-            )
+        if diffs := getattr(self, "differentials", None):
+            diffstr = f"\n (has differentials w.r.t.: {', '.join(map(repr, diffs))})"
```

Comment:
If we go with just calculating above, then also here,
```
diffstr = "\n (has differentials w.r.t.: " + ', '.join(map(repr, diffs)) + ")"
```

Reply:
I prefer the f-string.